fv/bpf_test.go

```diff
@@ -908,6 +908,46 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
 					})
 				})
 
+				Context("Test load balancer service with no backend", func() {
+					var testSvc *v1.Service
+					tgtPort := 8055
+					externalIP := []string{extIP}
+					srcIPRange := []string{}
+					testSvcName := "test-lb-service-extip"
+					var port uint16
+					var ip []string
+
+					BeforeEach(func() {
+						if testOpts.connTimeEnabled || testOpts.udpUnConnected {
```

Comment:
In case of UDP unconnected, its responsibility of the application to handle ICMP port unreachable, whereas in case of TCP and UDP-connected, the kernel responds with a "connection Refused". For UDP-unconnected, "test-connection" proceeds with send and does not get any data back and timesout. Hence UDP-unconnected is skipped.

Reply:
I think that's OK; the main point of the test is to verify that we're generating the ICMP message correctly. The fact that the kernel accepts it for both UDP and TCP seems pretty conclusive.

Reply:
OK, me surprised that works differently for the two versions of UDP, might be the libs, if it works, it works :)